src/Nri/Ui/Carousel/V2.elm

```diff
@@ -0,0 +1,175 @@
+module Nri.Ui.Carousel.V2 exposing
+    ( viewWithCombinedControls, viewWithPreviousAndNextControls
+    , viewWithTabControls, AriaLabel(..)
+    )
+
+{-| Patch changes:
+
+  - added new carousel APIs (with tabbed controls/previous and next controls, and combined controls)
+
+@docs viewWithCombinedControls, viewWithPreviousAndNextControls
+@docs viewWithTabControls, AriaLabel
+
+-}
+
+import Accessibility.Styled.Aria as Aria
+import Css exposing (..)
+import Html.Styled as Html exposing (..)
+import Html.Styled.Attributes as Attrs exposing (css)
+import TabsInternal.V2 as TabsInternal
+
+
+{-| Builds a carousel with previous and next controls
+Returns:
+`controls`: previous and next buttons element
+`slides` the container with the carousel contents
+`containerAttributes` attributes that should be used on the parent div of both the button and slides elements
+-}
+viewWithPreviousAndNextControls :
+    { selected : id
+    , panels :
+        List
+            { id : id
+            , slideHtml : Html msg
+            , ariaLabel : AriaLabel
+            }
+    , viewPreviousButton : Html msg
+    , viewNextButton : Html msg
+    , ariaLabel : AriaLabel
+    , controlListStyles : List Style
+    }
+    ->
+        { controls : Html msg
+        , slides : Html msg
+        , containerAttributes : List (Attribute msg)
+        }
+viewWithPreviousAndNextControls config =
+    { controls =
+        div [ css config.controlListStyles ]
+            [ config.viewPreviousButton, config.viewNextButton ]
+    , slides =
+        List.map
+            (\panel ->
+                Html.div
+                    [ Attrs.attribute "role" "group"
+                    , Aria.roleDescription "slide"
+                    , ariaLabelToAttr panel.ariaLabel
+                    , css
+                        [ if config.selected == panel.id then
+                            Css.display Css.block
+
+                          else
+                            Css.display Css.none
+                        ]
+                    ]
+                    [ panel.slideHtml ]
+            )
+            config.panels
+            |> Html.div [ Attrs.attribute "atomic" "false", Attrs.attribute "live" "polite" ]
+    , containerAttributes =
+        [ Attrs.attribute "role" "region"
+        , Aria.roleDescription "carousel"
+        , ariaLabelToAttr config.ariaLabel
+        ]
+    }
+
+
+ariaLabelToAttr : AriaLabel -> Attribute msg
+ariaLabelToAttr label =
+    case label of
+        IdLabel l ->
+            Aria.labeledBy l
+
+        StringLabel l ->
+            Aria.label l
+
+
+{-| Type which represents the type of aria label which will be used
+`IdLabel` will point to an existing element id on the DOM
+`StringLabel` will be a label of the element
+-}
+type AriaLabel
```

Comment:
🎨 thoughts on renaming to this...?

```elm
type LabelledBy
    = LabelledByIdOfVisibleLabel String
    | LabelledByAccessibleLabelOnly String
```

No big deal but I don't see us use "aria" anywhere in our APIs, and I think `AriaLabel` gives the impression that whatever we provide will end up in an `aria-label` attribute.